src/main/java/com/gladguys/alucoapi/helpers/GradeHelper.java

```diff
@@ -0,0 +1,29 @@
+package com.gladguys.alucoapi.helpers;
+
+import com.gladguys.alucoapi.entities.dto.ExamGradeDTO;
+
+import java.util.List;
+
+public class GradeHelper {
+
+	public static Double getAverageGrade(List<ExamGradeDTO> examGradesDTO) {
+
+		return getSumGrades(examGradesDTO) / getSumWeights(examGradesDTO);
+	}
+
+	private static int getSumWeights(List<ExamGradeDTO> examGradesDTO) {
+		return examGradesDTO
+					.parallelStream()
+					.map(ExamGradeDTO::getWeight)
+					.reduce(Integer::sum).get();
+	}
+
+	private static Double getSumGrades(List<ExamGradeDTO> examGradesDTO) {
+		return examGradesDTO
+					.parallelStream()
+					.filter(e -> e.getGrade() != null)
```

Comment:
Se for fresco que nem eu acho que pode fazer assim:

.filter(Objects::nonNull)